gateway/ln-gateway/tests/integration_tests.rs

```diff
@@ -51,11 +52,11 @@ async fn gatewayd_shows_info_about_all_connected_federations() {
     assert!(info
         .federations
         .iter()
-        .any(|info| info.federation_id == id1));
+        .any(|info| info.federation_id == id1 && info.balance_msat == Amount::ZERO));
     assert!(info
         .federations
         .iter()
-        .any(|info| info.federation_id == id2));
+        .any(|info| info.federation_id == id2 && info.balance_msat == Amount::ZERO));
 }
 
 #[tokio::test(flavor = "multi_thread")]
```

Comment:
On the `gatewayd_shows_balance_for_any_connected_federation` test case, I've started making an attempt, but not sure if this is the right direction to test using fixtures.

```rust
#[tokio::test(flavor = "multi_thread")]
async fn gatewayd_shows_balance_for_any_connected_federation() {
    let (_, rpc, fed1, fed2, bitcoin) = fixtures::fixtures().await;

    let id1 = fed1.connection_code().id;
    let id2 = fed2.connection_code().id;

    connect_federations(&rpc, &[fed1, fed2]).await.unwrap();

    let deposit_address1 = rpc.get_deposit_address(DepositAddressPayload { federation_id: id1 }).await.unwrap();
    let deposit_address2 = rpc.get_deposit_address(DepositAddressPayload { federation_id: id2 }).await.unwrap();

    // at least one epoch needed to establish fees
    bitcoin.prepare_funding_wallet().await;

    bitcoin.send_and_mine_block(&deposit_address1, bitcoin::util::amount::Amount::from_sat(10000)).await;
    bitcoin.send_and_mine_block(&deposit_address2, bitcoin::util::amount::Amount::from_sat(30000)).await;

    let info = rpc.get_info().await.unwrap();

    assert_eq!(info.federations.len(), 2);
    assert!(info
        .federations
        .iter()
        .any(|info| info.federation_id == id1 && info.balance_msat == Amount::from_sats(10000)));
    assert!(info
        .federations
        .iter()
        .any(|info| info.federation_id == id2 && info.balance_msat == Amount::from_sats(30000)));
}
```

I'm getting an error at the moment, which makes me believe either the fixture doesn't have a wallet module set up or I'm just totally on the wrong path here:

```shell
test gatewayd_shows_balance_for_any_connected_federation ... FAILED

failures:

---- gatewayd_shows_balance_for_any_connected_federation stdout ----
thread 'tokio-runtime-worker' panicked at 'No modules found of kind wallet', /Users/martinkang/personal/src/fedimint/fedimint-client/src/lib.rs:604:32
```

Could use some guidance!

Reply:
I'm not super familiar with these tests, but it seems you need to add more modules to your fixtures. So take a look at https://github.com/fedimint/fedimint/blob/66e90e8f17fa14949681c5b99a95fe51d5ee7e05/modules/fedimint-wallet-tests/tests/tests.rs#L17

and chain that using `.with_module(...)` on https://github.com/fedimint/fedimint/blob/66e90e8f17fa14949681c5b99a95fe51d5ee7e05/gateway/ln-gateway/tests/fixtures.rs#L25

Reply:
Cool, thanks. This is helpful. I'll give this a try and learn some more about the architecture and testing setup.

Reply:
I ended up working on this exact test concurrently.

Did you get this working? And, tangentially, when calling `rpc.get_balance`, I was not able to easily determine what the returned value represents. Does this represent balance as described on the btc blockchain, or does this represent balance in the lightning channel?

Reply:
I got past the wallet module issue, but still working on verifying the balance is reflected correctly.

I believe the return value represents the total amount of eCash notes in the federation.

Reply:
> the total amount of eCash notes in the federation

Interesting. So theoretically this could be an arbitrary number assigned by the guardians/consensus members, unrelated to the actual wealth held by the on-bitcoin multisig utxo.

There should probably be a comment somewhere explaining what that balance value means. There should also probably be some methods to fetch different lightning channel balances, and identify the bitcoin on-chain balance, right?